stdlib/REPL/src/REPL.jl

```diff
@@ -73,6 +73,22 @@ mutable struct REPLBackend
         new(repl_channel, response_channel, in_eval)
 end
 
+function softscope!(ex)
+    if ex isa Expr
+        h = ex.head
+        if h === :toplevel
+            for i = 1:length(ex.args)
+                ex.args[i] = softscope!(ex.args[i])
+            end
+        elseif h in (:meta, :import, :using, :export, :module, :error, :thunk)
+            return ex
+        else
+            return Expr(:block, Expr(:softscope), ex)
```

Comment:
Should this be something like `Expr(:softscope, true)`, with a corresponding `Expr(:softscope, false)` that turns it off?

I'm wondering whether there might ever be circumstances, e.g. in a macro, where you want to turn off softscope for a nested block even if the surrounding code is softscoped.